### DISCUSSION TITLE
Questions about PartsTreeStore

### DISCUSSION THREAD

**Discussion by ives9638** [tensorbase/tensorbase#125] [General]
hi @jinmingjian ，very interested  

After learning the basic design and concepts of Part
original description ：
**"Copa (column partition): TB is columnar, so the partition is columnar, called CoPa (short for Column Partition). The data in CoPa is only appended and unordered."**

the question:
If the partition data is only appended and unordered, the read will be expanded, thereby reducing query efficiency

In addition, if the data is directly dumped, disk space is also a problem.

**Comment by jinmingjian** [comment]
@ives9638 yes, you are right! You concerns will be solved soon in [this plan](https://github.com/tensorbase/tensorbase/blob/main/docs/summer_2021_ospp.md#summer-2021-of-open-source-promotion-plan).  The efficiency problem in fact also exists for LSM, and much worse. TensorBase is out of the ruins of the past. I personally think this become the new paradigm of columnar storage. Furthermore, it would be greater if you'd like to join our new effort. 🚀 😄

**Comment by jinmingjian** [comment]
We should have one RFC in some time recently. But you are interested in, we can discuss or start immediately. No need to wait any thing.

**Comment by jinmingjian** [comment]
current implementation has great room to improve, but it is prepared to challenge the sky-high of OLAP.

**Comment by jinmingjian** [comment]
some talked in issue #49
